File: server/src/main/java/com/sellis/server/model/SalonService.java
package com.sellis.server.model;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Document(collection = "services")
@CompoundIndex(name = "category_active_idx", def = "{'categoryId': 1, 'active': 1}")
public class SalonService {

    @Id
    private String id;

    private String name;

    @Indexed
    private String categoryId;

    private String categoryName;
    private String description;
    private Double price;
    private PriceRange priceRange;
    private List<ServiceVariant> variants = new ArrayList<>();
    private boolean hasVariants = false;
    private boolean active = true;
    private String imageUrl;

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;

    public SalonService() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getCategoryId() { return categoryId; }
    public void setCategoryId(String categoryId) { this.categoryId = categoryId; }
    public String getCategoryName() { return categoryName; }
    public void setCategoryName(String categoryName) { this.categoryName = categoryName; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public Double getPrice() { return price; }
    public void setPrice(Double price) { this.price = price; }
    public PriceRange getPriceRange() { return priceRange; }
    public void setPriceRange(PriceRange priceRange) { this.priceRange = priceRange; }
    public List<ServiceVariant> getVariants() { return variants; }
    public void setVariants(List<ServiceVariant> variants) { this.variants = variants; }
    public boolean isHasVariants() { return hasVariants; }
    public void setHasVariants(boolean hasVariants) { this.hasVariants = hasVariants; }
    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}

File: server-2/src/models/SalonService.js
const mongoose = require("mongoose");

const serviceVariantSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    price: { type: Number, required: true },
  },
  { _id: false }
);

const priceRangeSchema = new mongoose.Schema(
  {
    min: { type: Number, required: true },
    max: { type: Number, required: true },
  },
  { _id: false }
);

const salonServiceSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    categoryId: { type: String, required: true, index: true },
    categoryName: { type: String },
    description: { type: String },
    price: { type: Number },
    priceRange: { type: priceRangeSchema },
    variants: { type: [serviceVariantSchema] },
    hasVariants: { type: Boolean, default: false },
    active: { type: Boolean, default: true },
    imageUrl: { type: String },
  },
  { timestamps: true, collection: "services" }
);

salonServiceSchema.index({ categoryId: 1, active: 1 });

module.exports = mongoose.model("SalonService", salonServiceSchema);
